packages/select-v2/src/useSelect.ts

```diff
@@ -571,40 +585,65 @@ const useSelect = (props: ExtractPropTypes<typeof SelectProps>, emit) => {
     handleBlur()
   }
 
-  // in order to track these individually, we need to turn them into refs instead of watching the entire
-  // reactive object which could cause perf penalty when unnecessary field gets changed the watch method will
-  // be invoked.
+  const handleMenuEnter = () => {
+    states.inputValue = states.displayInputValue
+    nextTick(() => {
+      if (~indexRef.value) {
+        scrollToItem(indexRef.value)
+      }
+    })
+  }
 
-  watch(expanded, val => {
-    emit('visible-change', val)
-    if (val) {
-      popper.value.update?.()
-      // the purpose of this function is to differ the blur event trigger mechanism
-    } else {
-      states.displayInputValue = ''
-    }
-  })
+  const scrollToItem = (index: number) => {
+    menuRef.value.scrollToItem(index)
+  }
 
-  onMounted(() => {
+  const initStates = () => {
     if (props.multiple) {
       if ((props.modelValue as Array<any>).length > 0) {
+        states.cachedOptions.length = 0;
         (props.modelValue as Array<any>).map(selected => {
-          const item = props.options.find(option => option.value === selected)
+          const item = filteredOptions.value.find(option => option.value === selected)
           if (item) {
             states.cachedOptions.push(item as Option)
           }
         })
       }
     } else {
       if (props.modelValue) {
-        const selectedItem = props.options.find(o => o.value === props.modelValue)
+        const selectedItem = filteredOptions.value.find(o => o.value === props.modelValue)
 
         if (selectedItem) {
           states.selectedLabel = selectedItem.label
         }
       }
     }
+  }
+
+  // in order to track these individually, we need to turn them into refs instead of watching the entire
+  // reactive object which could cause perf penalty when unnecessary field gets changed the watch method will
+  // be invoked.
 
+  watch(expanded, val => {
+    emit('visible-change', val)
+    if (val) {
+      popper.value.update?.()
+      // the purpose of this function is to differ the blur event trigger mechanism
+    } else {
+      states.displayInputValue = ''
+    }
+  })
+
+  watch(() => props.modelValue, () => {
+    initStates()
+  })
+
+  watch(() => props.options, () => {
+    initStates()
+  })
```

Comment:
Should use

```
watch(expression, initStates)
```
?

Since we don't need any parameters here.

Reply:
Yeah, i merged them